### PR TITLE
fix(Core/Spell): Stack Enrage(arms) and Enrage(fury)

### PR DESCRIPTION
**Pull request by r0m1ntik**
## Changes Proposed:
-  fix stack for non stackable spell enrage([arms](https://wotlkdb.com/?spell=14204) and [fury](https://wotlkdb.com/?spell=57520)) all rank(1-5)

## Issues Addressed:
Close: [2473](https://github.com/azerothcore/azerothcore-wotlk/issues/2473)

## Tests Performed:
It's work, don't stack

## How to Test the Changes:

[[Stacking/Auras] Wrecking Crew and Enrage [Warrior]](https://github.com/azerothcore/azerothcore-wotlk/issues/2473)

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by FrancescoBorzi** [comment]
build failing @r0m1ntik , new startup error detected:

```
2021-02-22 19:59:53 Spell 12880 listed in `spell_group` has more than one group
2021-02-22 19:59:53 Spell 57518 listed in `spell_group` has more than one group
2021-02-22 19:59:53 SpellGroup id 1500 listed in `spell_group_stack_rules` does not exist
```

**Review by Helias — APPROVED** [review]

**Comment by claudiodfc** [comment]
## Tested performed
Have a Warrior with 4 points in **Wrecking Crew** and 5 points in **Enrage** fight Dragonblight NPCs.

## Test results
Whenever an NPC procs **Enrage** from the fury tree the **Wrecking crew** enrage effect gets removed and replaced by the latest. The same happens when you have **Enrage** talent from the fury tree active and deal a critical hit, activating **Wrecking Crew** from the Arms tree.

In the end, a 10% buff would be replaced by an 8% buff and viceversa which doesn't sound like the expected behaviour.

## Screenshots
WoWScrnShot_030421_142005
WoWScrnShot_030421_142009

**Comment by claudiodfc** [comment]
I just saw you added code for Death wish aswell. I went ahead and tested it and the same happened.
After casting _**Death Wish**_ and dealing a critical hit with, for example **Mortal Strike**, **_Wrecking Crew_** is activated and since I'm taking damage from NPCs _**Enrage**_ from the fury tree is once again activated.

WoWScrnShot_030421_143513

File: src/server/game/Spells/SpellInfo.cpp [hunk 1]
```diff
@@ -2133,6 +2133,10 @@ SpellSpecificType SpellInfo::LoadSpellSpecific() const
                         case 8115: // Agility
                         case 8091: // Armor
                             return SPELL_SPECIFIC_SCROLL;
+                        case 12880: // Enrage (Enrage)
+                        case 57518: // Enrage (Wrecking Crew)
+                        case 57514: // Enrage (Imp. Defensive Stance)
+                            return SPELL_SPECIFIC_WARRIOR_ENRAGE;
```
Comment (claudiodfc) [inline src/server/game/Spells/SpellInfo.cpp:2139]: This is a confusing change. 
I'm not sure this Warrior logic should be put on generic families. Maybe loading a spell correction and change it's family to SPELLFAMILY_WARRIOR would be wiser for future purposes.
Another thing that's bothering me is that you only added the Rank 1 Enrage spell for **Enrage** (12880) and the Rank 1 enrage for the spell 46867 - **Wrecking Crew (Rank 1)**. Which means that the other talent ranks will not be considered _SPELL_SPECIFIC_WARRIOR_ENRAGE_.

My suggestion would be
1. Finding where does the Modifier for damage is called and see how it's coded (probably on spell file)
2. Adding a priority list on each Enrage effect so one is not overridden by the other whenever procced.

Comment (Footman) [reply src/server/game/Spells/SpellInfo.cpp:2139]: 1. it is used with GetFirstRankSpell() that means he doesnt need to place every rank there.
2. Spellfamily from those Spell is generic in DBC. Why should he place it in Warrior family
3. I think replacing other proc by new one overridden proc is completely normal. At least it solves the problem

**Review by Footman — APPROVED** [review]

File: src/server/game/Spells/SpellInfo.cpp (hunk 1 above)
Comment (claudiodfc) [reply src/server/game/Spells/SpellInfo.cpp:2139]: You clearly didn't read my test report. But you do you

**Comment by temperrr** [comment]
@r0m1ntik Whats the status on this PR?

**Pull request closed without merging**